packages/bits/src/lib/wizard-v2/wizard-icon.directive.ts

```diff
@@ -0,0 +1,26 @@
+import {Directive, Input, TemplateRef} from "@angular/core";
+import {StepState} from "@angular/cdk/stepper";
+
+/** Template context available to an attached `NuiWizardIcon`. */
+export interface NuiWizardIconContext {
+    index: number;
+    active: boolean;
+    optional: boolean;
+}
+
+export interface MatStepperIconContext {
+    index: number;
+    active: boolean;
+    optional: boolean;
+}
+
+@Directive({
+    selector: "ng-template[nuiWizardIcon]",
+})
+export class NuiWizardIconDirective {
+
+    /** Name of the icon to be overridden. */
+    @Input() nuiWizardIcon: StepState;
```

Comment:
Maybe this instead?

```suggestion
    @Input() stepState: StepState;
```

Reply:
@Blakesalot Here I think it is a good idea to rename because this input is a step state in which icon should be replaced